### DISCUSSION TITLE
Integrate serving public key

### DISCUSSION THREAD

**Discussion by 3sv** [llamafilm/tesla-http-proxy-addon#150] [Ideas]
Is there any reason the flask server is not directly serving the public key for setup purposes? If the flask server would handle this task, the flow would be easier and less dependent on what software is used as webserver or ssl reverse proxy? The end user doesn't has to bother with the path ".well-known/appspecific/com.tesla.3p.public-key.pem" of that certificate as it is created internally already and can be served directly as well?

I used the docker version, and have a Caddy reverse proxy in front as reverse proxy to different services and for HTTPS termination. 
Created a config file to serve both Flask, the actual Tesla app, and the key in one go on the same endpoint. 
See also https://github.com/iainbullock/tesla-http-proxy-docker/issues/58 for details on how my setup is doing this at once, but for sure this would be possible from within the application as well and would make it easier for new users?

**Comment by llamafilm** [comment]
The public key must remain accessible at all times, it's not just for setup. I'm not an expert on Flask, but there are warnings all over its documentation that it should not be used in production. So I figured it would be safer to host directly with nginx. But you're right, it would be easier to setup and maybe it's fine being behind the reverse proxy.

**Comment by 3sv** [comment]
From what I tested it seems that the official Tesla part can handle serving the key at runtime. The path .well-known/appspecific/com.tesla.3p.public-key.pem is available, it requires an OAuth token to get to the public key, but that seems enough for actual using the API?

So if you would serve the key in the flask server during setup that would be enough and not really a security problem. Once the onboarding process is done, the flask server can still be disabled completely and the Tesla part can run at the same endpoint, removing the complex dependencies on configuration for the reverse proxy and keeping the same security level? Actually even improved security as the key is only served with an OAuth token in this case?

**Reply by llamafilm** [reply]
I don't know what you're talking about. There is no oauth on the key, it is _public_ by nature, and not served by the tesla-http-proxy at all.

Unless something changed since I first set this up, and my information is out of date. You're welcome to submit a PR.

**Reply by 3sv** [reply]
Just found backs in the the docs and it is indeed needed to keep the public key pem served directly.
Tested by serving only tesla-http-proxy, and then it is returning a json on the pem endpoint with a message that authentication is not filled in, so wasn't trivial to quickly test if it was serving the pem or not without a valid token. With that setup everything still worked from within HA correctly, but given the docs it will not keep working.

Guess only option to make it easier would be then to use something alike gunicorn to serve the file from Python. But I do understand that responsibilities are better separated as it is now.
